src/main/java/picard/illumina/parser/Range.java

```diff
@@ -45,12 +45,17 @@ public Range(final int start, final int end) {
 
     @Override
     public boolean equals(final Object object) {
-        if(object == null || !(object instanceof Range)) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
             return false;
         }
-
-        final Range that = (Range) object;
-        return that.start == this.start && that.end == this.end;
+        if (this.getClass() == object.getClass()) {
+            final Range that = (Range) object;
```

Comment:
This could cause a class cast exception since we don't check anywhere that "object" is an instance of Range.